schematic/store/synapse.py

```diff
@@ -659,105 +670,36 @@ def getFilesInStorageDataset(
         """
-        # select all files within a given storage dataset folder (top level folder in
-        # a Synapse storage project or folder marked with contentType = 'dataset')
-        walked_path = synapseutils.walk(
-            self.syn, datasetId, includeTypes=["folder", "file"]
-        )
+        file_list = []
 
-        current_entity_location = self.synapse_entity_tracker.get(
-            synapse_id=datasetId, syn=self.syn, download_file=False
-        )
+        # Identify all folders nested under the dataset folder
+        folders = synapseutils.walk(self.syn, datasetId, includeTypes=["folder"])
 
-        def walk_back_to_project(
-            current_location: Entity, location_prefix: str, skip_entry: bool
-        ) -> str:
-            """
-            Recursively walk back up the project structure to get the paths of the
-            names of each of the directories where we started the walk function.
-
-            Args:
-                current_location (Entity): The current entity location in the project structure.
-                location_prefix (str): The prefix to prepend to the path.
-                skip_entry (bool): Whether to skip the current entry in the path. When
-                    this is True it means we are looking at our starting point. If our
-                    starting point is the project itself we can go ahead and return
-                    back the project as the prefix.
-
-            Returns:
-                str: The path of the names of each of the directories up to the project root.
-            """
-            if (
-                skip_entry
-                and "concreteType" in current_location
-                and current_location["concreteType"] == PROJECT_ENTITY
-            ):
-                return f"{current_location.name}/{location_prefix}"
+        # Getting the files directly under the dataset will be the beginning of the query
+        dataset_clause = f"parentId='{datasetId}' "
 
-            updated_prefix = (
-                location_prefix
-                if skip_entry
-                else f"{current_location.name}/{location_prefix}"
-            )
-            if (
-                "concreteType" in current_location
-                and current_location["concreteType"] == PROJECT_ENTITY
-            ):
-                return updated_prefix
-            current_location = self.synapse_entity_tracker.get(
-                synapse_id=current_location["parentId"],
-                syn=self.syn,
-                download_file=False,
-            )
-            return walk_back_to_project(
-                current_location=current_location,
-                location_prefix=updated_prefix,
-                skip_entry=False,
-            )
+        # The query will also be ammended to include everything containted in all the subdirectories of the dataset
+        for subfolder, _, path in folders:
+            dataset_clause += f"OR parentId='{subfolder[1]}' "
```

Comment:
I did a test

```
import synapseclient
syn = synapseclient.login()
# AD Knowledge portal file view (254744 files/folders)
parents = syn.tableQuery("SELECT distinct parentId FROM syn11346063")
parents_df = parents.asDataFrame()
len(parents_df)
# 8905 folders

parents_str = "','".join(parents_df['parentId'].tolist())
total_query = syn.tableQuery(
    f"select * from syn11346063 where parentId in ('{parents_str}')"
)
temp = total_query.asDataFrame()
len(temp)
# 253744 files/folders
```

I would hope we don't go beyond 8000 folders?

Reply:
@thomasyu888 Thanks for the suggestion, I can see how that would be better. Are you also saying to get the folders to query from a query of the view too? Or was that just to facilitate the example here?
`path`, `id`, `name`, `parentId`, `projectId` and `type` I believe, though we should probably document that for users somewhere. These columns are all added by default now; `partentId` has been for a while and we've already been using it elsewhere in schematic for operations so this wouldn't be a new additional requirement

Reply:
@GiaJordan 
> Are you also saying to get the folders to query from a query of the view too? Or was that just to facilitate the example here?

That was just an example because that won't work without recursive SQL queries - I just wanted to get the MAX parentId's I could from a fileview.

Thanks - this is helpful, I'll add that to the documentation I've been working on.

Reply:
Okay, understood. And yeah no problem!